CMakeLists.txt

```diff
@@ -5,44 +5,51 @@
 # accordance with the terms of the Adobe license agreement accompanying
 # it. If you have received this file from a source other than Adobe,
 # then your use, modification, or distribution of it requires the prior
-# written permission of Adobe. 
+# written permission of Adobe.
 
-cmake_minimum_required(VERSION 3.18)
+cmake_minimum_required(VERSION 3.23)
+
+include(FetchContent)
+
+set(CMAKE_OSX_ARCHITECTURES x86_64) # Only build x86 for now
 
 project(hyde)
+
+if (NOT DEFINED CMAKE_CXX_STANDARD)
+    set(CMAKE_CXX_STANDARD 20)
+endif ()
+
 set(CMAKE_CXX_EXTENSIONS OFF)
-set(CMAKE_CXX_STANDARD 17)
 set(CMAKE_CXX_STANDARD_REQUIRED ON)
+set(CMAKE_XCODE_GENERATE_SCHEME OFF)
 
-if(APPLE)
-    # for now we are assuming the user has installed with homebrew.
-    # we detect the install location with brew --prefix llvm
-    execute_process(
-        COMMAND brew --prefix llvm
-        OUTPUT_VARIABLE llvm_to_use
-        OUTPUT_STRIP_TRAILING_WHITESPACE
+if (NOT TARGET llvm::llvm)
+    message(STATUS "Hyde third-party: creating target 'llvm::llvm'...")
+
+    FetchContent_Declare(
+        llvm
+        GIT_REPOSITORY https://github.com/llvm/llvm-project.git
+        GIT_TAG 8dfdcc7b7bf66834a761bd8de445840ef68e4d1a # llvmorg-15.0.7
+        GIT_SHALLOW TRUE
+        SOURCE_SUBDIR llvm
     )
-    message(STATUS "using this version of llvm: ${llvm_to_use}")
 
-    set(Clang_DIR ${llvm_to_use}/lib/cmake/clang/)
-    message(STATUS "Clang_DIR set to ${Clang_DIR}")
+    set(LLVM_ENABLE_PROJECTS clang)
+    set(LLVM_TARGETS_TO_BUILD X86)
 
-    set(LLVM_DIR ${llvm_to_use}/lib/cmake/llvm/)
-    message(STATUS "LLVM_DIR set to ${LLVM_DIR}")
+    FetchContent_MakeAvailable(llvm)
 endif()
 
-find_package(Clang REQUIRED clangTooling libClang clangASTMatchers)
-find_package(Clang REQUIRED CONFIG)
-find_package(LLVM REQUIRED CONFIG)
-
 set(Boost_USE_STATIC_LIBS ON)
-find_package(Boost COMPONENTS system filesystem REQUIRED)
 
-add_definitions(${Clang_DEFINITIONS})
-add_definitions(${LLVM_DEFINITIONS})
-add_definitions(${Boost_LIB_DIAGNOSTIC_DEFINITIONS})
```

Comment:
How about disabling autolinking entirely? From my perspective it is better to let the build system handle the link dependencies rather than relying on implicit pragmas. Boost has a `Boost::disable_autolinking` target for that, or you can define `BOOST_ALL_NO_LIB` directly.

Reply:
I've added BOOST_ALL_NO_LIB.